modules/tinymce/src/plugins/imagetools/main/ts/ui/Buttons.ts

```diff
@@ -5,55 +5,63 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Toolbar } from '@ephox/bridge';
+
 import Editor from 'tinymce/core/api/Editor';
 
 import * as Actions from '../core/Actions';
 
 const register = (editor: Editor) => {
   const cmd = (command: string) => () => editor.execCommand(command);
 
+  const onSetup = (buttonApi: Toolbar.ToolbarButtonInstanceApi) => {
```

Comment:
This is pretty inefficient, as is has to do this lookup at least 5 times for every `NodeChange` event. Can we please look at rewriting this so the lookup is only done once and then we notify any buttons that are setup?

The best way to do that is probably going to be to register the NodeChange event once and then have an array of listeners to call with the state. Let me know if you need more details or clarification on doing that.

- [ ] Optimize this lookup so it only needs to be done once

Reply:
I didn't write this part, I just moved what we already had. I'm happy to refactor but shouldn't it be another task?

Reply:
No, because previously it was only used for a single button so the logic was okay. You're now trying to use that same logic across multiple buttons.

Reply:
I tried to make the lookup more efficient using a map.